include/ddprof_process.hpp

```diff
@@ -35,28 +38,44 @@ class Process {
 
   uint64_t increment_counter() { return ++_sample_counter; }
 
+  [[nodiscard]] DwflWrapper *get_or_insert_dwfl();
+  [[nodiscard]] DwflWrapper *get_dwfl();
+  [[nodiscard]] const DwflWrapper *get_dwfl() const;
+
 private:
   static std::string format_cgroup_file(pid_t pid,
                                         std::string_view path_to_proc);
 
   static DDRes read_cgroup_ns(pid_t pid, std::string_view path_to_proc,
                               CGroupId_t &cgroup);
 
+  std::unique_ptr<DwflWrapper> _dwfl_wrapper{};
```

Comment:
```suggestion
  std::unique_ptr<DwflWrapper> _dwfl_wrapper;
```

Reply:
I did not have an opinion on this. Leaving as is unless we push for a different guideline. Feel free to challenge me further if this is important.